mozci/push.py

```diff
@@ -1037,15 +1037,14 @@ def classify_regressions(
         # - if a group is failing in all tasks, then it is a "cross-config" failure
         # - if a group is failing only in some tasks but not all, then it is not a "cross-config" failure
         all_groups = self.group_summaries.values()
+        failing_groups = [g for g in all_groups if g.status == Status.FAIL]
         groups_cross_config_failure = {
-            g.name for g in all_groups if g.is_cross_config_failure
```

Comment:
If we need to do this, it means the logic in https://github.com/mozilla/mozci/blob/211f65bfb1d698dfbef3d8661b0b330eb2ddd855/mozci/task.py#L430 is not correct.
We should fix that logic (add a few more tests for it too), and then the bug should automatically be fixed

Reply:
Indeed, if a group is passing we shouldn't be able to test if it's a cross config failure or not at all.
To prevent this, I'm refactoring the `is_cross_config_failure()` property to `cross_config_state()` that will output one of the 3 following states, `PASSED`, `CROSS_CONFIG` or `NOT_CROSS_CONFIG`.

Reply:
That was already done in the previous implementation, we were using `not is_cross_config_failure` hence the issue because we were getting not only the non cross config failures but also the groups that passed.

Reply:
Ah, I see the problem! Let's then change `groups_non_cross_config_failure` to be:
```
groups_non_cross_config_failure = {g.name for g in all_groups if g.status != Status.PASS and not g.is_cross_config_failure}
```

Reply:
done